packages/ui-library/src/components/icon-menu/index.tsx

```diff
@@ -63,13 +101,32 @@ const IconMenu = ({ options, color = 'info', variant = 'text', icon, size = 'sma
           horizontal: 'left',
         }}
       >
-        {options.map(({ onClick, icon: itemIcon, label, control, closeOnClick = true }) => (
-          <MenuItem key={label} onClick={() => handleCloseWithAction(closeOnClick, onClick)}>
-            {itemIcon}
-            {label}
-            {control}
-          </MenuItem>
-        ))}
+        {options.map(
+          ({
+            onClick,
+            icon: itemIcon,
+            label,
+            secondaryLabel,
+            control,
+            closeOnClick = true,
+            bottomDivider,
+            color: itemColor,
+          }) => (
+            <Box key={label}>
```

Comment:
I needed to render `<Divider />` alongside `<MenuItem />` as siblings. At first, I tried to use a React Fragment, but this was unsupported by Material-UI's Menu. Consequently, wrapping them in a plain div became the most viable solution, especially since each pair needed to be returned as a single component inside a map function.

Reply:
I have removed the Box Container now, by a condition based on the OptionType